packages/contracts/contracts/optimistic-ethereum/OVM/bridge/tokens/OVM_L2StandardTokenFactory.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >0.5.0 <0.8.0;
+pragma experimental ABIEncoderV2;
+
+/* Contract Imports */
+import { L2StandardERC20 } from "../../../libraries/standards/L2StandardERC20.sol";
+import { Lib_PredeployAddresses } from "../../../libraries/constants/Lib_PredeployAddresses.sol";
+
+/**
+ * @title OVM_L2StandardTokenFactory
+ * @dev Factory contract for creating standard L2 token representations of L1 ERC20s
+ * compatible with and working on the standard bridge.
+ * Compiler used: optimistic-solc
+ * Runtime target: OVM
+ */
+contract OVM_L2StandardTokenFactory {
+
+    event StandardL2TokenCreated(address indexed _l1Token, address indexed _l2Token);
+
+    /**
+     * @dev Creates an instance of the standard ERC20 token on L2.
+     * @param _l1Token Address of the corresponding L1 token.
+     * @param _name ERC20 name.
+     * @param _symbol ERC20 symbol.
+     */
+    function createStandardL2Token(
+        address _l1Token,
+        string memory _name,
+        string memory _symbol
+    )
+        external
+    {
+        require (_l1Token != address(0), "Must provide L1 token address");
+
+        L2StandardERC20 l2Token = new L2StandardERC20(
```

Comment:
Do all L2StandardERC20 tokens get 18 decimal places?

Reply:
Ah - doesn't seem worth changing this for flexibility, never mind. https://github.com/OpenZeppelin/openzeppelin-contracts/blob/6842518b1b71fac9a21c7d94ec521992cff266b5/contracts/token/ERC20/ERC20.sol#L86

Reply:
Yeah I've had to deploy custom token for WBTC because of this. But seems okay design-wise.